memgpt/agent.py

```diff
@@ -441,9 +444,21 @@ def _get_ai_reply(
         except Exception as e:
             raise e
 
-    def _handle_ai_response(self, response_message: Message, override_tool_call_id: bool = True) -> Tuple[List[Message], bool, bool]:
+    def _handle_ai_response(
+        self,
+        response_message: ChatCompletionMessage,  # TODO should we eventually move the Message creation outside of this function?
+        override_tool_call_id: bool = True,
+        # If we are streaming, we needed to create a Message ID ahead of time,
+        # and now we want to use it in the creation of the Message object
+        # TODO figure out a cleaner way to do this
+        response_message_id: Optional[str] = None,
```

Comment:
@sarahwooders FYI we are now passing `response_message_id` into `handle_ai_response` for the special case where we created the `Message` object before we started unpacking it / turning it into inner thoughts / actions / etc.